src/components/structures/auth/Registration.js

```diff
@@ -288,7 +288,16 @@ module.exports = React.createClass({
         });
     },
 
-    onFormValidationFailed: function(errCode) {
+    onFormValidationChange: function(fieldErrors) {
+        // Find the first error and show that.
+        const errCode = Object.values(fieldErrors).find(value => value);
```

Comment:
ooi why not `[0]`?

Reply:
There can be `null` values for fields without errors, so we need the first non-null one. Maybe `!!value` or a comment would make it clearer...

Reply:
js is also weird and doesn't break on `[0]` fwiw

```
> const testArray = [];
undefined
> testArray[0]
undefined
```

Reply:
Ah, but it's not about the values array being empty, but instead the element in it. For example, the values array might be `[null, "bob", null]`, and we want `"bob"` here.

Reply:
Oh, I see. Yup, that makes sense. A comment would be great :)